### PR TITLE
tsm v2

### PR DESCRIPTION
**Pull request by seriphdog**
###submit to github
1) has add to default_cases.txt
2) Pr Info:
2.1) performance in stat.csv:
name,shape,gops,time(ms),mac_utilization,cpu_usage,ddr_utilization
name,shape,gops,time(ms),mac_utilization,cpu_usage,ddr_utilization
tsm_bm1684x_Bf16,1x3x8x224x224,65.554,22.381,146.45%,6.00%,61.78%
tsm_bm1684x_f16,1x3x8x224x224,65.554,22.401,146.32%,6.00%,61.73%
tsm_bm1684x_f32,1x3x8x224x224,65.554,86.062,38.09%,11.00%,32.78%
compilation,1x3x8x224x224,65.554,22.406,N/A,6.00%,N/A
compilation,1x3x8x224x224,65.554,86.085,N/A,11.00%,N/A
compilation,1x3x8x224x224,65.554,22.403,N/A,6.00%,N/A
fp32:flops: 65554824720, runtime: 62.249783ms, ComputationAbility: 1.053093T
fp16:flops: 65554824720, runtime: 21.997709ms, ComputationAbility: 2.980075T
bf16:flops: 65554824720, runtime: 21.997709ms, ComputationAbility: 2.980075T
2.2)tpu-mlir:
branch:master
commit id: 315ccc9a5ba3b401039e399fd6f8cdc92f70c78b
2.3)libsophon:0.4.5

### CODE REVIEW

**Comment by seriphdog** [comment]
tsm pr && tsm v1 can be delete

**Comment by XiangBin-gh** [comment]
run_cali如果没问题就打开

**Comment by seriphdog** [comment]
> run_cali如果没问题就打开
run_cali 得到的table有问题，具体在一层slice切成2个Local片，第一个slice片的min=max=thre=0会触发assert，故注释掉。

**Review by Jeffwhen — CHANGES_REQUESTED** [review]

File: dataset/tiny-kinetics-400/tsm_cali_table
Comment (Jeffwhen) [inline dataset/tiny-kinetics-400/tsm_cali_table:1]: 放模型目录中 data 目录

File: dataset/tiny-kinetics-400/top_output_tsm.npz
Comment (Jeffwhen) [inline dataset/tiny-kinetics-400/top_output_tsm.npz:1]: 放模型目录中 data 目录

File: dataset/tiny-kinetics-400/tiny_kinetic_calibration/1.npy
Comment (Jeffwhen) [inline dataset/tiny-kinetics-400/tiny_kinetic_calibration/1.npy:1]: 放模型目录中 data 目录

File: dataset/tiny-kinetics-400/input_tsm.npz
Comment (Jeffwhen) [inline dataset/tiny-kinetics-400/input_tsm.npz:1]: 非通用数据，放模型目录中 data 目录

**Review by Jeffwhen — CHANGES_REQUESTED** [review]

File: vision/recognition/tsm/README.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+|---------------|--------------------------------------------------------------|
+|Proposed       |TSM_kinetics_RGB_resnet50_shift8_blockres_avg_segment8_e50.pth|
+
+## Dataset
+
+We sugguest use tiny dataset(<https://github.com/Tramac/tiny-kinetics-400>) rather than large Kinetic-400.
+
+## References
+
+None
```
[the remark this answers is not in the document]
Comment (seriphdog) [reply vision/recognition/tsm/README.md]: 已加入工程链接

File: vision/recognition/tsm/README.md [hunk 2]
```diff
@@ -0,0 +1,26 @@
+<!--- SPDX-License-Identifier: Apache-2.0 -->
+
+# TSM
+
+## Description
+
+This is a implementation of the Temporal Shift Module described in the paper
```
[the remark this answers is not in the document]
Comment (seriphdog) [reply vision/recognition/tsm/README.md]: 补充模型描述

File: vision/recognition/tsm/tsm.mlir.config.yaml [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    --scale 0.01712475383,0.0175070028,0.01742919389
+
+# mlir_calibration:
+#   run_calibration.py $(workdir)/$(name).mlir
+#     --dataset $(tiny_kinetic_dataset)/tiny_kinetic_calibration/
+#     --input_num 1
+#     -o $(workdir)/$(name)_cali_table
+
+deploy:
+  model_deploy.py
```
[the remark this answers is not in the document]
Comment (seriphdog) [reply vision/recognition/tsm/tsm.mlir.config.yaml]: 所有model_transform.py都加了-

File: vision/recognition/tsm/README.md [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+
+## Model
+
+|Model          |Instance                                                      |
+|---------------|--------------------------------------------------------------|
+|Proposed       |TSM_kinetics_RGB_resnet50_shift8_blockres_avg_segment8_e50.pth|
+
+## Dataset
+
+We sugguest use tiny dataset(<https://github.com/Tramac/tiny-kinetics-400>) rather than large Kinetic-400.
```
[the remark this answers is not in the document]
Comment (seriphdog) [reply vision/recognition/tsm/README.md]: 已修改

File: vision/recognition/tsm/README.md [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+## Description
+
+This is a implementation of the Temporal Shift Module described in the paper
+["TSM: Temporal Shift Module for Efficient Video Understanding"](https://arxiv.org/abs/1811.08383).
+
+## Model
+
+|Model          |Instance                                                      |
+|---------------|--------------------------------------------------------------|
+|Proposed       |TSM_kinetics_RGB_resnet50_shift8_blockres_avg_segment8_e50.pth|
```
[the remark this answers is not in the document]
Comment (seriphdog) [reply vision/recognition/tsm/README.md]: 已修改成示例

File: dataset/tiny-kinetics-400/top_output_tsm.npz
Comment (seriphdog) [reply dataset/tiny-kinetics-400/top_output_tsm.npz:1]: 数据全部换到了tsm/data下面

File: dataset/tiny-kinetics-400/tiny_kinetic_calibration/1.npy
Comment (seriphdog) [reply dataset/tiny-kinetics-400/tiny_kinetic_calibration/1.npy:1]: 数据全部换到了tsm/data下面

File: dataset/tiny-kinetics-400/input_tsm.npz
Comment (seriphdog) [reply dataset/tiny-kinetics-400/input_tsm.npz:1]: 数据全部换到了tsm/data下面

File: vision/recognition/tsm/README.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (seriphdog) [reply vision/recognition/tsm/README.md]: paper链接已放入下方inference

**Comment by seriphdog** [comment]
githubstatus已正常，已rebase

**Comment by Jeffwhen** [comment]
rebase 成一笔，full_cases.list 有冲突不用管，最后合入时候处理下就行。

File: dataset/tiny-kinetics-400/tsm_cali_table
Comment (seriphdog) [reply dataset/tiny-kinetics-400/tsm_cali_table:1]: done

**Comment by seriphdog** [comment]
> rebase 成一笔，full_cases.list 有冲突不用管，最后合入时候处理下就行。

Done

**Pull request merged**